### PR TITLE
xmlexport improve compat with xml schema

### PR DESCRIPTION
**Pull request by bitkeeper**
Improve compatibility with:
* https://opcfoundation.org/UA/2011/03/UANodeSet.xsd
* https://opcfoundation.org/UA/2008/02/Types.xsd
Impove this compatibility make it possible to read the information models into other tools like UaModeler.

Mostly fixes to garantee the order of the elements according the xsd:
* NamespaceUris: Make sure that the namespacuris are on top (above the aliases)
* UAVariable: Place the values below the references
* uax:Boolean:  values should be 'true' and 'false' (lower case)
* Argument: place the members in the correct order

Btw correct format of the xml file can be checked with XML validator against the XSD.

### CODE REVIEW

**Comment by oroulet** [comment]
Thanks! I am really not an xml export, do feel free to explain what is
wrong. Maybe you can add automatic test to check generated files with xsd
in tests?

File: opcua/common/xmlexporter.py [hunk 1]
```diff
@@ -404,8 +407,20 @@ def _extobj_to_etree(self, val_el, name, dtype, val):
...
-            self.member_to_etree(struct_el, name, ua.NodeId(getattr(ua.ObjectIds, vtype)), getattr(val, name))
+        items_keys = val.ua_types.keys()
+        skip_empty = False
+        if dtype == ua.NodeId(ua.ObjectIds.Argument):
+            skip_empty = True
+            items_keys = [name for name in ['Name',
+                                            'DataType',
+                                            'ValueRank',
+                                            'ArrayDimensions',
+                                            'Description'] if name in items_keys ]
```
Comment (oroulet) [inline opcua/common/xmlexporter.py]: Why do we need to hard code the order? That seems strange....

Comment (bitkeeper) [inline opcua/common/xmlexporter.py]: That isn't so strange: Just like `Argument.to_binary` also requires an fixed order of the members, the xmls schema (https://opcfoundation.org/UA/2008/02/Types.xsd) also requires an fixed order.

In a lot of cases the order doesn't matter, but if a tool checks the import against the scheme the xml parse will fail.

**Comment by oroulet** [comment]
But it is a lot of code for one of 1000 cases, isn't it any logic in that
ordering? Maybe we can use the order of nodeids?

**Comment by oroulet** [comment]
Ok we do not have nodeids here...

**Comment by bitkeeper** [comment]
There are indeed a lot more cases, but using arguments (for methods) is that common. Now we have at least a working version for the most basic stuff. 

If more cases are required to be fixed (read really used by others) an order lookup will be required to prevent a code bloat. But lets only invest that time when it is required, I sure that we both have better things to at the moment ... 

The current matching FrozenClass currently contains a dictionary, so it will also not provide the order  information.

Maybe if the matching FrozenClass would contain the order of the member it would be possible to automate:
* `to_binary`
* `_binary_init`
* `__str__`
* `__init__`
* xml serialize order

**Comment by zerox1212** [comment]
I have thought about this a few times, and have been wondering if it would be smart to refactor the entire library to use OrderedDict. It isn't very pythonic, but the UA spec seems to assume things like classes will have fixed orders.

We have the same issue with custom structures mentioned here after much research. https://github.com/FreeOpcUa/python-opcua/issues/373

**Comment by oroulet** [comment]
Ordereddict is supposed to be a drop in replacement to dict, so it should
not be hard to use, but it may impact performance and I do really see where
it would help...

**Comment by bitkeeper** [comment]
@oroulet To get this pr accepted: 
Should I revert the changes from this pr concerning the order of Argument members ?
The order fixes in this pr are indepent from it.

**Comment by oroulet** [comment]
If you can split it, it would be great. I am unsure about the Argument ordering. It is a lot of code for something which is quite suspect..

**Comment by bitkeeper** [comment]
Ok reverted the part about the member order of the Argument. Nevertheless at least the implementation of the member order of the `Argument `is wrong. Still an other implementation is required for that.

**Comment by oroulet** [comment]
Make another pr for the order stuff
 Maybe we'll find a generic solution...

**Pull request merged**